modules/ROOT/pages/authorization-grant-types.adoc

```diff
@@ -0,0 +1,103 @@
+
+* Provides the access token to access the protected resource.
+
+The Implicit grant type optimizes performance for browser clients that use a scripting language, such as Javascript. However, this grant type can expose the access token to the resource owner or the resource owner's user-agent.
+
+An OAuth dance that uses an Implicit grant type works as follows:
+
+. Client app calls the service provider's authorization server, providing its client ID to confirm that it is a registered client of the service provider.
+
+. Client app uses a login page supplied by the service provider to ask that the resource owner authenticate itself to the service provider.
+
+. After confirming that the client ID is valid (that is, the client app had previously registered with the service provider), and authenticating the resource owner's login credentials, the service provider returns an access token.
+
+. Client app calls the resource server, providing the access token, to request the protected resource.
+
+. Service provider delivers the protected resource.
+
+
+== Resource Owner Password Credentials
+
+With the Resource Owner Password Credentials grant type, a client application demands that the resource owner share their service provider login credentials. The client app then uses these credentials to access the service provider and log into the resource owner's account. The service provider accepts the valid login credentials and grants the client app full access to the resource owner's account.
+
+This type of authorization optimizes performance at the expense of sharing secure access credentials. The resource owner must trust the client app not to share access to private data.
+
+An OAuth dance uses the Resource Owner Password Credentials grant type works as follows:
+
+. Resource owner provides access credentials to the client app.
+
+. Client app calls the service provider to request an access token. During the call, the client provides the client ID to confirm that it is a registered client of the service provider, and also provides the access credentials for the protected resource.
+
+. Service provider authenticates the client app and validates the access credentials. Then it issues an access token.
+
+. Client app calls the resource server, providing the access token, to request access to the protected resource.
+
+. Service provider delivers the protected resource.
+
+== Client Credentials
+
+With the Client Credentials grant type, a client app requests access to data that is not associated with a particular resource owner. Because no protected resources are involved, a service provider must only verify that the client as valid. This type of OAuth interaction is sometimes referred to as "two-legged OAuth" because it involves only two roles: the service provider and a client app.  The other three authorization types involve a third role, the resource owner. Thus, they are sometimes referred to as "three-legged OAuth".
```

Comment:
"no protected resources" -> maybe "no password-protected resources" or something like that? Looks like it's a case where authentication is not required, only authorization. Is that correct?

Reply:
OK